### PR TITLE
Add soak test workflow.

### PR DESCRIPTION
**Pull request by anuraaga**
Ok for review but let me merge since it has dependency on https://github.com/open-telemetry/opentelemetry-lambda/pull/48

### CODE REVIEW

**Comment by anuraaga** [comment]
Only difference vs integration tests workflow is

1) names use git sha
2) Delete layers from terraform before destruction if soak test passes

**Comment by anuraaga** [comment]
As a followup we would add https://github.com/aws-observability/aws-otel-lambda/pull/19/files#diff-93dcdebadcc088e25be16bdc7166019b0b5b58a35d5cef13c97ba78909814632 in

File: .github/workflows/soaking.yml [hunk 1]
```diff
@@ -1,160 +1,139 @@
...
-        if: ${{ success() }}
+          registry: public.ecr.aws
+      - name: Build layers / functions
+        run: ${{ matrix.build_command }}
+        working-directory: ${{ matrix.build_directory }}
+      - name: Build collector
+        run: make package
+        working-directory: opentelemetry-lambda/collector
+      - uses: hashicorp/setup-terraform@v1
+      - name: Initialize terraform
```
Comment (wangzlei) [inline .github/workflows/soaking.yml:112]: we can merge terraform init/apply into one step, to reduce the code length.

File: .github/workflows/soaking.yml [hunk 2]
```diff
@@ -1,160 +1,139 @@
...
+        working-directory: ${{ matrix.terraform_directory }}
+      - name: Apply terraform
+        run: terraform apply -auto-approve
+        working-directory: ${{ matrix.terraform_directory }}
+        env:
+          # TODO(anuraaga): Consider picking a canonical language to manage the pushed collector layer and use different
+          # names for others. Won't be needed if we don't publish separate collector layer.
+          TF_VAR_collector_layer_name: aws-otel-lambda-collector-${{ github.sha }}
+          TF_VAR_sdk_layer_name: aws-otel-lambda-${{ layer_type }}-${{ github.sha }}
+          TF_VAR_function_name: hello-lambda-${{ matrix.language }}-${{ github.run_id }}-${{ matrix.name }}
```
Comment (wangzlei) [inline .github/workflows/soaking.yml:123]: Do you want to support both SAM and TF in soak and build-main workflows? or TF only?

**Pull request merged**